lms/static/js/edxnotes/views/tab_view.js

```diff
@@ -122,10 +122,14 @@ define([
          * Shows error message.
          */
         showErrorMessage: function (message) {
-            this.$('.wrapper-msg')
-                .removeClass('is-hidden')
-                .find('.msg-content .copy').html(message);
+            var $wrapper = this.$('.wrapper-msg');
+            $wrapper.removeClass('is-hidden');
+
+            if (!(message instanceof HtmlUtils.HtmlSnippet)) {
```

Comment:
It is dangerous because now a caller could do:

  showErrorMessage(rawHtmlFromUser)

and that rawHtml could have XSS scripts in it. The code as it is currently defined will decide that it isn't an HtmlSnippet and would then call ``HtmlUtils.HTML`` on it which is declaring that the HTML is safe, even though this method cannot tell whether it is safe. There is no way for the linter to detect this, because it has to assume that methods that call ``HtmlUtils.HTML`` trust the string being passed in.

If a method isn't sure if a parameter is safe, then it should always call ``HtmlUtils.ensureHtml`` on it to verify that it is before passing it to JQuery. If the caller of this method knows that it has safe HTML that it doesn't want double escaped then it shouldn't pass it as a string, but instead should use one of the ``HtmlUtils`` methods to pass it through as a snippet.